lib/util.js

```diff
@@ -29,13 +29,21 @@ function rp (options) {
     });
 }
 
-function fetch_html(uri) {
+function fetch_html(uri, encoding=null) {
     uri = encode_uri(uri);
 
-    return rp({
+    const options = {
         uri: uri,
         gzip: true,
-    }).then(res => {
+    };
+
+    // Allow user to override encoding for instances where a site isn't giving
+    // the proper content-type
+    if (encoding) {
+        options.encoding = encoding;
+    }
```

Comment:
Since it's defaulted to null in the param list, can't you just set encoding in the options above to:

```
const options = {
  uri: uri,
  encoding: encoding,
  gzip: true,
};
```